includes/constants/class-base-constant.php

```diff
@@ -103,4 +103,14 @@ public static function __callStatic( $name, $arguments ) {
 	public function __toString() {
 		return constant( \get_class( $this ) . '::' . $this->get_value() );
 	}
+
+	/**
+	 * Specify the value which should be serialized to JSON.
+	 *
+	 * @return mixed|string
+	 */
+	#[\ReturnTypeWillChange]
+	public function jsonSerialize() {
+		return $this->__toString();
+	}
```

Comment:
I'd add a simple test assertion for this method:
 
```php
		$this->assertSame( wp_json_encode( $class ), '"bacs_debit"' );
```

after this line 
https://github.com/Automattic/woocommerce-payments/blob/7f7a83e8b31edd783ddea83407994754f6b82100/tests/unit/test-class-base-constant.php#L19

Reply:
👍 I've added a test inline with this suggestion here: 270c587eb9750d4777c724b9db014af2e6cdbe3b

Reply:
@james-allan - it seems that the test is failing. I think it's important to have double quotes bounding the string as it's the result of JSON serialziation. 

```
1) Base_Constant_Test::test_base_constant_will_create_constant
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'"bacs_debit"'
+'bacs_debit'
```

Reply:
ah yes. I copied the test above and forgot to update the expected value to include those quotes. I've added that in [2c06324](https://github.com/Automattic/woocommerce-payments/pull/6753/commits/2c063244e8c1bc2299fb13b01d9dfbf300612f9d). Thanks Dat.